sync/_databricks.py

```diff
@@ -885,16 +1026,41 @@ def _get_job_cluster(tasks: List[dict], job_clusters: list) -> Response[dict]:
     return Response(error=DatabricksError(message="Not all tasks use the same cluster"))
 
 
-def _get_run_cluster_id(tasks: List[dict]) -> Response[str]:
-    cluster_ids = {task["cluster_instance"]["cluster_id"] for task in tasks}
-    num_ids = len(cluster_ids)
+def _get_cluster_id_and_tasks_from_run_tasks(
+    all_tasks: List[dict],
+    exclude_tasks: Union[Collection[str], None] = None,
+    project_id: str = None,
+) -> Tuple[str, List[dict]]:
+    cluster_id_to_tasks = {}
+    for task in all_tasks:
+        if not exclude_tasks or task["task_key"] not in exclude_tasks:
+            cluster_id = task["cluster_instance"]["cluster_id"]
+            if cluster_id not in cluster_id_to_tasks:
+                cluster_id_to_tasks[cluster_id] = [task]
+            else:
+                cluster_id_to_tasks[cluster_id].append(task)
+
+    num_ids = len(cluster_id_to_tasks)
+    if num_ids == 0:
+        raise Exception("No cluster found for tasks")
+    elif num_ids > 1:
+        raise Exception("More than 1 cluster found for tasks")
```

Comment:
Can we move this to after we optionally filter by project ID? That way we can target clusters within a job.